### ISSUE TITLE
MA0011 not reported when type does not implement IFormattable

### ISSUE THREAD

**Issue by drieseng** [meziantou/Meziantou.Analyzer#909]
### Version of the Meziantou.Analyzer NuGet package

2.0.243

### Rule Identifier

MA0011

### Target Framework

.NET 9.0

### C# Language version

4.14.0-3.25412.6 (c167adef)

### Description

When a given type does not implement **IFormattable** and has both a `ToString()` and a `ToString(IFormatProvider)`, then **MA0011** is not reported when `ToString()` is invoked.

### Reproduction Steps

Minimal code:

```c#
using System;
using System.Globalization;

internal sealed class Program
{
    private static void Main()
    {
        Location location = new()
            {
                Start = 1,
                End = 5
            };
        Console.WriteLine(location.ToString());
    }
}

public sealed class Location
{
    public required int Start { get; init; }

    public required int End { get; init; }

    public override string ToString()
    {
        return ToString(CultureInfo.CurrentCulture);
    }

    public string ToString(IFormatProvider formatProvider)
    {
        return string.Empty;
    }
}
```

**Comment by drieseng** [comment]
Even if confirmed that it's a bug, I'm ok with not fixing this issue if there's a big impact on performance,
In which case, it's enough to document this as such.
